src/mobu/jupyterclient.py

```diff
@@ -29,46 +34,93 @@ class NotebookException(Exception):
     pass
 
 
-@dataclass
+class JupyterClientSession:
+    """Wrapper around `aiohttp.ClientSession` using token authentication.
+
+    Unfortunately, aioresponses does not capture headers set on the session
+    instead of with each individual call, which means that we can't see the
+    token and thus determine what user we should interact with in the test
+    suite.  Work around this with this wrapper class around
+    `aiohttp.ClientSession` that just adds the token header to every call.
+
+    Parameters
+    ----------
+    session : `aiohttp.ClientSession`
+        The session to wrap.
+    token : `str`
+        The token to send.
+    """
+
+    def __init__(self, session: ClientSession, token: str) -> None:
+        self._session = session
+        self._token = token
+
+    async def close(self) -> None:
+        await self._session.close()
+
+    def request(
+        self, method: str, url: str, **kwargs: Any
+    ) -> _RequestContextManager:
+        if "headers" not in kwargs:
+            kwargs["headers"] = {}
+        kwargs["headers"]["Authorization"] = f"Bearer {self._token}"
+        return self._session.request(method, url, **kwargs)
+
+    def delete(self, url: str, **kwargs: Any) -> _RequestContextManager:
+        return self.request("delete", url, **kwargs)
+
+    def get(self, url: str, **kwargs: Any) -> _RequestContextManager:
+        return self.request("get", url, **kwargs)
+
+    def post(self, url: str, **kwargs: Any) -> _RequestContextManager:
+        return self.request("post", url, **kwargs)
+
+    def ws_connect(
+        self, *args: Any, **kwargs: Any
+    ) -> _WSRequestContextManager:
+        if "headers" not in kwargs:
+            kwargs["headers"] = {}
+        kwargs["headers"]["Authorization"] = f"Bearer {self._token}"
+        return self._session.ws_connect(*args, **kwargs)
+
+
 class JupyterClient:
-    log: BoundLoggerLazyProxy
-    user: User
-    session: ClientSession
-    headers: Dict[str, str]
-    xsrftoken: str
-    jupyter_url: str
-
-    def __init__(
-        self, user: User, log: BoundLoggerLazyProxy, options: Dict[str, Any]
-    ):
+    """Client for talking to JupyterHub.
```

Comment:
Hub/Lab: the Configurable Hub Proxy will connect it to the lab post-authentication in the general case, and for things like code execution we are talking to the lab, not the Hub.

One of the most confusing things for me about the whole mobu implementation has been "does this method talk to the Hub, or (via the CHP) to the lab?" as well as "When I say 'session' do I mean aiohttp session or do I mean JL session?"